Boost: Backport 4.5.3 Readme changes

LiamSarsfield opened the pull request:
<!--- This template is used for backporting changes made during a plugin release back to trunk. -->

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
Backports readme.txt change for Boost 4.5.3

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
N/a
## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No
## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

Proofread changes.

LiamSarsfield committed:
Changelog updates

LiamSarsfield committed:
Changelog updates

LiamSarsfield committed:
Composer.lock

LiamSarsfield committed:
Update stable tag in readme.txt to version 4.5.3

LiamSarsfield committed:
Merge remote-tracking branch 'origin/trunk' into update/boost/backport-4.5.3

dilirity reviewed (dismissed):
Thanks for taking care of it!

:shipit:

LiamSarsfield committed:
Add changelog entry for boost backport 4.5.3 update

This commit introduces a new changelog entry to document a minor change in readme.txt, ensuring consistency with the SVN version.
